### PR TITLE
Feature: Show multiple languages 

### PR DESCRIPTION
**Pull request by haddadzineddine**
## Description

Modify the user interface to include support for displaying ontologies data in multiple languages.

Follow up : #239 

## Screenshots
Screenshot 2023-05-23 131050

## Changes made : 

- Refactor ontology_viewer_page_name to support multiple languages #3e1fde56
- Refactor sort_by in collection and schemes to support multiple languges #0b9282c6
- Update ontology_viewer title #fc4d801d
- Add all option in language selection #24ad1b31
- Show prefLabel with the specific language in case of all languages #52eed1f3

## Reviewers
@syphax-bouazzouni

### CODE REVIEW

**Comment by syphax-bouazzouni** [comment]
Add to this PR, the display of the treeview depending on the selected language, like the mockup below 
image

**Comment by haddadzineddine** [comment]
image

**Comment by syphax-bouazzouni** [comment]
> image

Don't take always the first, do a selection by portal language preference. if the portal languages are  `:en`, and `:fr`, then show English first if found, else French otherwise show what you have randomly.

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
Don't work

image

**Comment by haddadzineddine** [comment]
Screenshot 2023-06-08 161159

**Comment by haddadzineddine** [comment]
@syphax-bouazzouni

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -186,14 +188,27 @@ def tree_link_to_concept(child:, ontology_acronym:, active_style:, node: nil)
     icons = child.relation_icon(node)
     muted_style = child.isInActiveScheme&.empty? ? 'text-muted' : ''
     href = ontology_acronym.blank? ? '#' : "/ontologies/#{child.explore.ontology.acronym}/concepts/?id=#{CGI.escape(child.id)}&language=#{language}"
+
+    if child.prefLabel.nil?
+      prefLabelHTML =  child.id.split('/').last
+    else
+      prefLabelLang, prefLabelHTML = get_concept_label(child.prefLabel)
```
Comment (syphax-bouazzouni) [inline app/helpers/application_helper.rb:195]: extract a helper here

File: app/helpers/collections_helper.rb [hunk 2]
```diff
@@ -25,14 +25,24 @@ def get_collections_labels(collections, main_uri = '')
...
+      label = select_language_label(get_collection_label(x))
       if id.eql? main_uri
         selected_label = { 'prefLabel' => label, '@id' => id }
       else
         collections_labels.append( { 'prefLabel' => label, '@id' => id , 'color' => x['color'] })
       end
     end
-    collections_labels.sort_by! { |s|  s['prefLabel']}
+
+    collections_labels.sort_by! do |s|
```
Comment (syphax-bouazzouni) [inline app/helpers/collections_helper.rb:36]: extract a re-usable helper here

File: app/helpers/ontologies_helper.rb [hunk 3]
```diff
@@ -467,13 +467,17 @@ def language_selector_hidden_tag(section)
   end
 
   def languages_options(submission =  @submission || @submission_latest)
-    current_lang = request_lang
+    current_lang = request_lang.downcase
     submission_lang = submission_languages(submission)
```
Comment (syphax-bouazzouni) [inline app/helpers/ontologies_helper.rb:471]: here handle the case, where submission_languages is empty.

File: app/helpers/ontologies_helper.rb [hunk 4]
```diff
@@ -467,13 +467,17 @@ def language_selector_hidden_tag(section)
   end
 
   def languages_options(submission =  @submission || @submission_latest)
-    current_lang = request_lang
```
Comment (syphax-bouazzouni) [inline app/helpers/ontologies_helper.rb:470]: here handle the case when request_lang is nil, select the portal language

File: app/helpers/schemes_helper.rb [hunk 5]
```diff
@@ -24,15 +24,24 @@ def get_schemes_labels(schemes, main_uri)
...
+        label[1] = "#{label[1]} (main)" unless label[0].empty?
         selected_label = { 'prefLabel' => label, '@id' => id }
       else
         schemes_labels.append( { 'prefLabel' => label, '@id' => id })
       end
     end
-    schemes_labels.sort_by! { |s|  s['prefLabel']}
+
+    schemes_labels.sort_by! do |s|
+      pref_label = s['prefLabel']
```
Comment (syphax-bouazzouni) [inline app/helpers/schemes_helper.rb:37]: extract a re-usable helper (same as collection)

File: app/helpers/schemes_helper.rb [hunk 6]
```diff
@@ -82,21 +91,36 @@ def schemes_data
 
   def tree_link_to_schemes(schemes_labels, main_scheme_label, selected_scheme_id)
     out = ''
-    schemes_labels.sort_by { |s| [s['prefLabel']] }.each do |s|
+   
+    schemes_labels.sort_by { |s|
+      pref_label = s['prefLabel']
```
Comment (syphax-bouazzouni) [inline app/helpers/schemes_helper.rb:96]: duplicate with line 37

File: app/views/ontologies/sections/properties.html.haml [hunk 7]
```diff
@@ -5,50 +5,7 @@
 
   :plain
     <script id="property-details" type="text/x-handlebars-template">
-      <table cellpadding="0" cellspacing="0" class="minimal concept_details" width="100%">
```
Comment (syphax-bouazzouni) [inline app/views/ontologies/sections/properties.html.haml:8]: why this is removed ?

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
I refactored some of the code and pointed out some changes to make.
@haddadzineddine  please test it again and see the changes requested.

File: app/helpers/ontologies_helper.rb (hunk 4 above)
Comment (syphax-bouazzouni) [reply app/helpers/ontologies_helper.rb:315]: for now by default  portal language is set to "EN" in request_lang helper

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**